Much faster scan, checks only top offers and other changes

AqlaSolutions opened the pull request:
Other changes: checks only offers of enough amount, fixed logging for non-english ship names, fixed missing commander.defaultorder error in log.

To use improved scan speed update existing settings - set scan speed to at least 100.

Sorry for having it all in one pull request, as always I wasn't thought to publish my changes until there was too many of them. If you don't want to bother looking into all my changes I understand that.

DeadAirRT commented:
I'll start looking over the changes

Just from my initial impression, having a hard forced minimum amount of 80% is a bad idea and will potentially lead to a lot of downtime depending on the economy. Also, there are many times where a trade of 50% capacity is more profit than a trade of 100% capacity. It could definitely work as an option the person can set though but I'd like to develop the idea a bit more to where minamount gets relaxed if the trader can't find anything for a while.

Thank you for identifying the issue with ship names for debug, I didn't realize it had issues with non-english names.

I don't mind the scan settings personally, but I have to basically make it so that certain users can't break it.

I'll keep looking through this though and think about some of the things.

AqlaSolutions commented:
The big reason I added `minAmount` was "checks only top offers" optimization which uses `relativeprice` sorter. For Drone Parts I had many offers with <10 amount which were sorted to the top offers. This caused other more profitable offers with less `relativeprice` to be ignored because they were at the bottom of the list. MinAmount filtered out all these 3-4 amount offers and fixed this issue. It should work well even with 30% though. Also minAmount filter in the outer loop (buy offers) drastically improves speed because for lower amounts it doesn't need to take inner loop (sell offers) at all.

AqlaSolutions commented:
Also for scan speed: in my game it literally took minutes to scan at maximum speed of 15. Now on speed 100 with other changes the scan takes <15 seconds and lowers my fps a bit but not noticeable. And after I limited wares to most traded and profitable it takes just a few seconds. Using too low scan speed causes more cpu power to be wasted on switching to wait and back instead of actually processing offers.

AqlaSolutions commented:
I have more changes based on this, see here https://github.com/AqlaSolutions/TaterTrader/commit/e6bcad50037e3bd09ce820a1f6437893316a40b2
